Library/Homebrew/cmd/update.sh

```diff
@@ -331,6 +345,13 @@ EOS
     UPSTREAM_BRANCH="$(upstream_branch)"
     # the refspec ensures that the default upstream branch gets updated
     (
+      if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
+      then
+        # Skip taps without formulae.
+        FORMULAE="$(find "$DIR" -maxdepth 1 -name '*.rb' -or -name 'Formula' -print -quit)"
+        [ -z "$FORMULAE" ] && exit
```

Comment:
```sh
[[ -z "$FORMULAE" ]] && exit
```

Ideally, also make this variable local and name it `formulae`, even though it's not going to escape the subshell we're executing this part of the code in.